lib/_stream_transform.js

```diff
@@ -217,3 +230,110 @@ function done(stream, er, data) {
     throw new ERR_TRANSFORM_ALREADY_TRANSFORMING();
   return stream.push(null);
 }
+
+function SourceIterator(asyncGeneratorFn, opts) {
```

Comment:
can we use `class SourceIterator` ?

Reply:
see https://github.com/nodejs/node/pull/28501/files#r299247252

Reply:
Using only class syntax without private properties shouldn't make backporting more difficult

Reply:
yes - but the second point is - the surrounding context is also constructor style, happy to convert all of it to classes but that's a bigger picture problem

Reply:
I don't feel strongly about it. I think we haven't changed existing code because of backwards-compatibility.

Reply:
yeah, it feels like a bigger discussion - I started out writing it as class syntax and converted to constructor in respect for the context. If there's some other guiding principal, then I'm also happy to adjust to that. It would be good to know what the consensus is on this so it doesn't yo-yo between paradigms.

Reply:
resolving this since it seems there isn't (yet) a broad consensus on approach. If there's a mandate to convert code es6 classes I'd definitely be happy to convert this and all streams code in a separate PR